aws/data_source_aws_glue_script.go

```diff
@@ -0,0 +1,182 @@
+package aws
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsGlueScript() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsGlueScriptRead,
+		Schema: map[string]*schema.Schema{
+			"dag_edge": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"source": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"target": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"target_parameter": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"dag_node": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"args": {
+							Type:     schema.TypeList,
+							Required: true,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"param": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									"value": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"line_number": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"node_type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"language": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  glue.LanguagePython,
+				ValidateFunc: validation.StringInSlice([]string{
+					glue.LanguagePython,
+					glue.LanguageScala,
+				}, false),
+			},
+			"python_script": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"scala_code": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```

Comment:
I'm not sure why the Glue API distinguishes between the return values. It only returns one of them at a time currently 😕  Maybe it will return multiple in the future. We tend to opt for sticking with API rather than making our own choices for UX due to past maintenance issues. I'm tempted to leave this as-is _hoping_ that AWS has a good reason for this API design decision.